File: python/compare-locales/compare_locales/tests/test_util.py
# This Source Code Form is subject to the terms of the Mozilla Public
# License, v. 2.0. If a copy of the MPL was not distributed with this
# file, You can obtain one at http://mozilla.org/MPL/2.0/.

import unittest

from compare_locales import util


class ParseLocalesTest(unittest.TestCase):
    def test_empty(self):
        self.assertEquals(util.parseLocales(''), [])

    def test_all(self):
        self.assertEquals(util.parseLocales('''af
de'''), ['af', 'de'])

    def test_shipped(self):
        self.assertEquals(util.parseLocales('''af
ja win mac
de'''), ['af', 'de', 'ja'])

    def test_sparse(self):
        self.assertEquals(util.parseLocales('''
af

de

'''), ['af', 'de'])

File: mobile/android/android-components/components/browser/storage-sync/src/test/java/mozilla/components/browser/storage/sync/TestUtil.kt
/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

package mozilla.components.browser.storage.sync

import android.content.Context
import androidx.test.core.app.ApplicationProvider
import androidx.work.testing.WorkManagerTestInitHelper
import mozilla.components.service.glean.Glean
import java.util.Date

/**
 * Enable testing mode, initialize [WorkManagerTestInitHelper] for testing,
 * and initialize Glean. This is used in tests that send Glean pings.
 *
 * @param context the application context to init Glean with
 */
internal fun resetGlean(
    context: Context = ApplicationProvider.getApplicationContext()
) {
    Glean.enableTestingMode()

    // We're using the WorkManager in a bunch of places, and Glean will crash
    // in tests without this line. Let's simply put it here.
    WorkManagerTestInitHelper.initializeTestWorkManager(context)

    Glean.initialize(context)
}

fun Date.asSeconds() = time / BaseGleanSyncPing.MILLIS_PER_SEC
